presto-kudu/src/test/java/io/prestosql/plugin/kudu/TestKuduIntegrationDynamicFilter.java

```diff
@@ -0,0 +1,206 @@
+        {
+            return CompletableFuture.runAsync(() -> {
+                try {
+                    TimeUnit.HOURS.sleep(1);
+                }
+                catch (InterruptedException e) {
+                    throw new IllegalStateException(e);
+                }
+            });
+        }
+
+        @Override
+        public boolean isComplete()
+        {
+            return false;
+        }
+
+        @Override
+        public boolean isAwaitable()
+        {
+            return true;
+        }
+
+        @Override
+        public TupleDomain<ColumnHandle> getCurrentPredicate()
+        {
+            return TupleDomain.all();
+        }
+    }
+
+    @Test
+    public void testJoinDynamicFilteringSingleValue()
+    {
+        // Join lineitem with a single row of orders
+        assertDynamicFiltering(
+                "SELECT * FROM lineitem JOIN orders ON lineitem.orderkey = orders.orderkey AND orders.comment = 'nstructions sleep furiously among '",
+                withBroadcastJoin(),
+                6,
+                ImmutableSet.of(6, 1));
+    }
+
+    @Test
+    public void testJoinDynamicFilteringBlockProbeSide()
+    {
+        // Wait for both build sides to finish before starting the scan of 'lineitem' table (should be very selective given the dynamic filters).
+        assertDynamicFiltering(
+                "SELECT l.comment" +
+                        " FROM  lineitem l, part p, orders o" +
+                        " WHERE l.orderkey = o.orderkey AND o.comment = 'nstructions sleep furiously among '" +
+                        " AND p.partkey = l.partkey AND p.comment = 'onic deposits'",
+                withBroadcastJoinNonReordering(),
+                1,
+                ImmutableSet.of(1));
+    }
+
+    private void assertDynamicFiltering(@Language("SQL") String selectQuery, Session session, int expectedRowCount, Set<Integer> expectedOperatorRowsRead)
+    {
+        DistributedQueryRunner runner = (DistributedQueryRunner) getQueryRunner();
+        ResultWithQueryId<MaterializedResult> result = runner.executeWithQueryId(session, selectQuery);
+
+        assertEquals(result.getResult().getRowCount(), expectedRowCount);
+        assertEquals(getOperatorRowsRead(runner, result.getQueryId()), expectedOperatorRowsRead);
+    }
+
+    private Session withBroadcastJoin()
+    {
+        return Session.builder(this.getQueryRunner().getDefaultSession())
+                .setSystemProperty(JOIN_DISTRIBUTION_TYPE, BROADCAST.name())
+                .build();
+    }
+
+    private Session withBroadcastJoinNonReordering()
+    {
+        return Session.builder(this.getQueryRunner().getDefaultSession())
+                .setSystemProperty(JOIN_DISTRIBUTION_TYPE, BROADCAST.name())
+                .setSystemProperty(JOIN_REORDERING_STRATEGY, NONE.name())
+                .build();
+    }
+
+    private static Set<Integer> getOperatorRowsRead(DistributedQueryRunner runner, QueryId queryId)
```

Comment:
we changed that method to return list now: `io.prestosql.plugin.memory.TestMemorySmoke#getOperatorRowsRead`